### ISSUE TITLE
addon-docs: "No props found for this component" when component is exported in camelCase

### ISSUE THREAD

**Issue by ranndev** [storybookjs/storybook#9575]
**Describe the bug**

Props table is empty when the default exported variable of the component file was in the camel case format.

**Code snippets**

Here is my MDX file:

*icon-button.stories.mdx*
```
import { Meta, Props } from '@storybook/addon-docs/blocks';
import IconButton from './icon-button';

<Meta
  component={IconButton}
  title="IconButton"
/>

## Props

<Props of={IconButton} />
```

And here is my component:

**Not working when the default exported variable is "iconButton"**

*icon-button.tsx*
```typescript
import PropTypes from 'prop-types';
import React, { FC } from 'react';
import './icon-button.scss';

export interface IProps {
  /**
   * button color
   */
  color?: string;
}

const iconButton: FC<IProps> = function IconButton(props) {
  return <div className="icon-button">icon-button</div>;
};

iconButton.propTypes = {
  color: PropTypes.string,
};

iconButton.defaultProps = {
  color: 'primary',
};

export default iconButton;
```

**Not it works after I changed the default exported variable to "iconbutton"**

*icon-button.tsx*
```typescript
import PropTypes from 'prop-types';
import React, { FC } from 'react';
import './icon-button.scss';

export interface IProps {
  /**
   * button color
   */
  color?: string;
}

const iconbutton: FC<IProps> = function IconButton(props) {
  return <div className="icon-button">icon-button</div>;
};

iconbutton.propTypes = {
  color: PropTypes.string,
};

iconbutton.defaultProps = {
  color: 'primary',
};

export default iconbutton;
```

**Screenshots**

![2020-01-21 12-46-32 2020-01-21 12_48_57](https://user-images.githubusercontent.com/24478046/72776260-87b21f00-3c4c-11ea-8b9e-28f44f1dd623.gif)

**Expected behavior**
Should still work with camelCase

**System:**

  System:
    OS: macOS Mojave 10.14.4
    CPU: (4) x64 Intel(R) Core(TM) i7-4578U CPU @ 3.00GHz
  Binaries:
    Node: 10.16.3 - /usr/local/bin/node
    Yarn: 1.21.1 - ~/.yarn/bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
  Browsers:
    Chrome: 79.0.3945.130
    Firefox: 70.0.1
    Safari: 12.1
  npmPackages:
    @storybook/addon-a11y: ^5.3.7 => 5.3.7 
    @storybook/addon-actions: ^5.3.7 => 5.3.7 
    @storybook/addon-docs: ^5.3.7 => 5.3.7 
    @storybook/addon-jest: ^5.3.7 => 5.3.7 
    @storybook/addon-knobs: ^5.3.7 => 5.3.7 
    @storybook/addons: ^5.3.7 => 5.3.7 
    @storybook/core: ^5.3.7 => 5.3.7 
    @storybook/react: ^5.3.7 => 5.3.7

**Comment by Zunaib** [comment]
you need to do export const abc (if in camelCase)

**Comment by shilman** [comment]
Repro: https://github.com/storybookjs/storybook/commit/03a269399981849c2b2c4c256bb345b4b9829e65

**Comment by sidhuko** [comment]
Experiencing a similar problem. Use `export const` with TypeScript and have docgen setup and other components working. Some reason `React.FC<SomeProps>` won't be identified.

**Comment by thany** [comment]
> 
> 
> OP could you try changing your code to use `import * as React from 'react';`?

This fixes some situation for me. Why does this matter???

**Comment by sidhuko** [comment]
Some issue in: https://www.npmjs.com/package/storybook-addon-react-docgen

A contributor said it they will be removing for the react-docgen package directly in Storybook v6. There are still caveats using my fix. On other issues I think they have made suggestions on how to implement react-docgen in v5.

**Comment by shilman** [comment]
Correct. We'll be recommending react-docgen in SB 6.0. Still working out the details!

**Comment by shilman** [comment]
Details updated: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#react-prop-tables-with-typescript

**Comment by shilman** [comment]
A few weeks ago we released zero-config typescript support in 6.0-beta, and then a few days ago upgraded `react-docgen-typescript-loader` to `react-docgen-typescript-plugin`. This will solve many typescript props-related issues, possibly including this one:

- [Release 6.0.0-beta.26+](https://github.com/storybookjs/storybook/releases/tag/v6.0.0-beta.12)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#zero-config-typescript)
- [Updated docs](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/configurations/typescript-config/index.md)

Please try it out!
